nhais-adaptor/adaptor/outgoing/message_adaptor.py

```diff
@@ -0,0 +1,136 @@
+        :return: MessageSegmentPatientDetails
+        """
+        gender_map = {'unknown': 0, 'male': 1, 'female': 2, 'other': 9}
+
+        patient = odh.find_resource(fhir_operation, resource_type=ResourceType.PATIENT)
+
+        edi_name = MessageAdaptor.create_patient_name(patient.name[0])
+
+        edi_address = MessageAdaptor.create_patient_address(patient.address[0])
+
+        msg_seg_patient_details = MessageSegmentPatientDetails(id_number=patient.identifier[0].value,
+                                                               name=edi_name,
+                                                               date_of_birth=patient.birthDate.as_json(),
+                                                               gender=gender_map[patient.gender],
+                                                               address=edi_address)
+        return msg_seg_patient_details
+
+    @staticmethod
+    def create_message_segment_registration_details(fhir_operation):
+        """
+        Create the message segment registration details from the fhir operation
+        :param fhir_operation:
+        :return: MessageSegmentRegistrationDetails
+        """
+
+        transaction_number = odh.get_parameter_value(fhir_operation=fhir_operation,
+                                                     parameter_name=ParameterName.TRANSACTION_NO)
+
+        if fhir_operation.name == OperationName.REGISTER_BIRTH:
+            acceptance_code = "A"
+            acceptance_type = "1"
+
+        practitioner_details = odh.find_resource(fhir_operation=fhir_operation, resource_type=ResourceType.PRACTITIONER)
+        party_id = f"{practitioner_details.identifier[0].value},{practitioner_details.identifier[1].value}"
+
+        patient_details = odh.find_resource(fhir_operation=fhir_operation, resource_type=ResourceType.PATIENT)
+        birth_location = patient_details.extension[0].valueAddress.city
+
+        msg_seg_registration_details = MessageSegmentRegistrationDetails(transaction_number=transaction_number,
+                                                                         party_id=party_id,
+                                                                         acceptance_code=acceptance_code,
+                                                                         acceptance_type=acceptance_type,
+                                                                         date_time=fhir_operation.date.as_json(),
+                                                                         location=birth_location)
+        return msg_seg_registration_details
+
+    @staticmethod
+    def create_message_beginning(fhir_operation):
+        """
+        Create the beginning of the message
+        :param fhir_operation:
+        :return: MessageBeginning
+        """
+        nhais_id = odh.get_parameter_value(fhir_operation=fhir_operation, parameter_name=ParameterName.NHAIS_CYPHER)
+
+        if fhir_operation.name == OperationName.REGISTER_BIRTH:

```

Comment:
This doesn't look like it'll be very flexible when we look to expand to supporting other operations. Is this class specific to the birth registration message?

Reply:
I've added a dictionary to map these codes based upon the operation name. Should make this more flexible going forward